modules/service/src/main/scala/lucuma/odb/service/AsterismService.scala

```diff
@@ -97,6 +98,24 @@ object AsterismService {
 
     new AsterismService[F] {
 
+      // Selects the program id that is shared by all the given observations,
+      // or else fails.
+      private def selectProgramId(
+        observationIds: NonEmptyList[Observation.Id]
+      ): F[Result[Program.Id]] = {
+        val af = Statements.selectProgramId(observationIds)
+        session.prepareR(af.fragment.query(program_id)).use { p =>
+          p.stream(af.argument, chunkSize = 16)
```

Comment:
Could you just use `unique` instead of stream? Or is it just easier to handle the error case like this?

Reply:
Yeah, that was my first attempt but I wasn't happy with the error handling.  I guess it could go either way.

Reply:
No strong opinion here. This is fine.

Reply:
I want it to fail unless there is exactly one result.  `unique` would fail with an exception when more than one is returned by the query or if none are returned, which I could have `recoverWith`ed I think.  That might be clearer I'm not sure.

Reply:
Gotcha, yeah either way works. What you have is fine.

Reply:
Thanks.  I went back and tried again with the `unique` / `recover` option and remembered the reason I switched was that I would have to catch `skunk.exception.SkunkException` which seemed overly broad. I'll just leave it this way

```
skunk.exception.SkunkException: Exactly one row was expected, but more were returned.
	at skunk.exception.SkunkException$.fromQueryAndArguments(SkunkException.scala:145)
	at skunk.PreparedQuery$$anon$1.fail(PreparedQuery.scala:134)
	at skunk.PreparedQuery$$anon$1.unique$$anonfun$1(PreparedQuery.scala:118)
	at cats.effect.IOFiber.succeeded(IOFiber.scala:1204)
	at cats.effect.IOFiber.runLoop(IOFiber.scala:245)
	at cats.effect.IOFiber.asyncContinueSuccessfulR(IOFiber.scala:1372)
	at cats.effect.IOFiber.run(IOFiber.scala:113)
	at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:634)
```